cpupower/cpupower.py

```diff
@@ -0,0 +1,181 @@
+        """
+        cmd_governor_list = "cat /sys/devices/system/cpu/cpu%s/cpufreq/scaling_available_governors" % (self.i)
+        output = process.system_output(cmd_governor_list)
+        governors = output.split()
+        return governors
+
+    def get_freeq(self):
+        """
+        Get the cur,min,max freequency on the system
+        """
+        cmd1 = "cat /sys/devices/system/cpu/cpu%s/cpufreq/scaling_min_freq" % (self.i)
+        cmd2 = "cat /sys/devices/system/cpu/cpu%s/cpufreq/scaling_max_freq" % (self.i)
+        cmd3 = "cat /sys/devices/system/cpu/cpu%s/cpufreq/scaling_cur_freq" % (self.i)
+        min = process.system_output(cmd1).strip()
+        max = process.system_output(cmd2).strip()
+        cur = process.system_output(cmd3).strip()
+        return (min, max, cur)
+
+    def set_governor(self, governor):
+        """
+        Set the governor specified in the argument
+        """
+        cmd = "cpupower frequency-set -g %s" % (governor)
+        output = process.run(cmd)
+        cmd = "cat /sys/devices/system/cpu/cpu%s/cpufreq/scaling_governor" % (self.i)
+        cur_governor = process.system_output(cmd).strip()
+        if (output.exit_status == 0) and (governor == cur_governor):
+            self.log.info("%s governor set successfully" % governor)
+        else:
+            self.log.error("%s governor set failed" % governor)
+            self.nfail += 1
+
+    def get_cur_freq(self):
+        """
+        Get the current frequency info on system
+        """
+        cmd1 = "cpupower -c %s frequency-info -f" % (self.i)
+        output1 = process.system_output(cmd1).splitlines()[1]
+        freq = output1.split()[3]
+        return freq
+
+    def get_random_freq(self):
+        """
+        Get random freequency from list
+        """
+        cmd = "cat /sys/devices/system/cpu/cpu%s/cpufreq/scaling_available_frequencies" % (self.i)
+        value = process.system_output(cmd)
+        value2 = []
+        value2 = value.split(' ')
+        rand = random.choice(value2)
+        return rand
+
+    def set_freq_val(self, freq):
+        """
+        Set the freequency value specified in argument
+        """
+        cmd = "cpupower frequency-set -f %s" % (freq)
+        output = process.run(cmd)
+        cur_freq = self.get_cur_freq()
+        if (output.exit_status == 0) and (cur_freq == freq):
+            self.log.info("the userspace governor working as expected and freeq set correctly")
+        else:
+            self.log.error("Userspace governor failed")
+            self.nfail += 1
+
+    def check_performance_governor(self, governor, max):
+        """
+        Validate Performance Governor
+        """
+        self.set_governor(governor)
+        cur_freq = self.get_cur_freq()
+        if cur_freq == max:
+            self.log.info("%s governor working as expected" % governor)
+        else:
+            self.log.error("%s governor not working as expected" % governor)
+            self.nfail += 1
```

Comment:
Have a template for values to be checked for different governors in a dict  and use common function(check_governor) for all governors that would help in future incase of additional governors to be added.